src/lib/swap/HtlcUtils.js

```diff
@@ -0,0 +1,168 @@
+            redeemAddress: recipient,
+            hash: hashRoot,
+            timeoutBlockHeight: timeout,
+        };
+    }
+
+    /**
+     * @param {unknown} script - Uint8Array
+     * @returns {BtcHtlcContents}
+     */
+    static decodeBtcHtlcScript(script) {
+        const error = new Errors.InvalidRequestError('Invalid BTC HTLC script');
+
+        if (!script || !(script instanceof Uint8Array) || !script.length) throw error;
+        const chunks = BitcoinJS.script.decompile(/** @type {Buffer} */ (script));
+        if (!chunks) throw error;
+        const asm = BitcoinJS.script.toASM(chunks).split(' ');
+
+        let branchesVerifiedIndividually = false;
+
+        /* eslint-disable no-plusplus */
+        let i = 0;
+
+        // Start redeem branch
+        if (asm[i] !== 'OP_IF') throw error;
+
+        // Check secret size
+        if (asm[++i] !== 'OP_SIZE' || asm[++i] !== (32).toString(16) || asm[++i] !== 'OP_EQUALVERIFY') throw error;
+
+        // Check hash
+        if (asm[++i] !== 'OP_SHA256' || asm[i + 2] !== 'OP_EQUALVERIFY') throw error;
+        const hash = Nimiq.BufferUtils.toHex(Nimiq.BufferUtils.fromHex(asm[++i]));
+        ++i;
+
+        // Check redeem address
+        if (asm[++i] !== 'OP_DUP' || asm[++i] !== 'OP_HASH160') throw error;
+        const redeemAddressBytes = Nimiq.BufferUtils.toHex(Nimiq.BufferUtils.fromHex(asm[++i]));
+
+        // End redeem branch, start refund branch
+        if (asm[++i] !== 'OP_ELSE') {
+            branchesVerifiedIndividually = true;
+            if (asm[i] !== 'OP_EQUALVERIFY' || asm[++i] !== 'OP_CHECKSIG' || asm[++i] !== 'OP_ELSE') throw error;
+        }
+
+        // Check timeout
+        // @ts-ignore Argument of type 'Buffer' is not assignable to parameter of type 'Buffer'
+        const timeoutTimestamp = BitcoinJS.script.number.decode(BitcoinJS.Buffer.from(asm[++i], 'hex'));
+        if (asm[++i] !== 'OP_CHECKLOCKTIMEVERIFY' || asm[++i] !== 'OP_DROP') throw error;
+
+        // Check refund address
+        if (asm[++i] !== 'OP_DUP' || asm[++i] !== 'OP_HASH160') throw error;
+        const refundAddressBytes = Nimiq.BufferUtils.toHex(Nimiq.BufferUtils.fromHex(asm[++i]));
+
+        // End refund branch
+        if (branchesVerifiedIndividually) {
+            if (asm[++i] !== 'OP_EQUALVERIFY' || asm[++i] !== 'OP_CHECKSIG' || asm[++i] !== 'OP_ENDIF') throw error;
+        } else {
+            // End contract
+            // eslint-disable-next-line no-lonely-if
+            if (asm[++i] !== 'OP_ENDIF' || asm[++i] !== 'OP_EQUALVERIFY' || asm[++i] !== 'OP_CHECKSIG') throw error;
+        }
+
+        if (asm.length !== ++i) throw error;
+        /* eslint-enable no-plusplus */
+
+        return {
+            refundAddressBytes,
+            redeemAddressBytes,
+            hash,
+            timeoutTimestamp,
+        };
+    }
+
+    /**
+     * @param {Buffer[]} witness
+     * @returns {Buffer}
+     */
+    static witnessStackToScriptWitness(witness) {
```

Comment:
This is a re-implementation of a method included in BitcoinJS, which is sadly not exported anywhere.

https://github.com/bitcoinjs/bitcoinjs-lib/blob/533d6c2e6d0aa4111f7948b1c12003cf6ef83137/ts_src/psbt.ts#L1512